### PR TITLE
MM-34278 Creating Resource share for Enterprise TGWs and IAM role

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
- Resource share for Enterprise TGWs 
- IAM role genesis assumption to add principals

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34278

### CODE REVIEW

File: terraform/aws/modules/genesis-support/variables.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+variable "share_name_test" {}
```
Comment (spirosoik) [inline terraform/aws/modules/genesis-support/variables.tf:1]: just a preference, it's to have a descriptions in variables so we can get in a first glance the need of them. also it's nice each module to have it's own document. There is a nice tool which generates everything in markdown and leverages the description of variables and default values too https://github.com/terraform-docs/terraform-docs

**Review by spirosoik — APPROVED** [review]
one minor comment to consider for the future

File: terraform/aws/modules/genesis-support/variables.tf (hunk 1 above)
Comment (stylianosrigas) [reply terraform/aws/modules/genesis-support/variables.tf:1]: @spirosoik Yes I agree on that. We keep all the details in the main templates and not the actual modules but we need to start giving some love on the open source modules. I will start by adding description to this module here and later we can see about the document thing.

**Review by angeloskyratzakos — APPROVED** [review]
LGTM

**Pull request merged**